### PR TITLE
[FINCC-3466] Add Billing Address to Payment Method Create Request

### PR DESCRIPTION
**Pull request by saopayne**
**Ticket**
https://hellofresh.atlassian.net/browse/FINCC-3466

**Description:**
For PaymentMethod::Create and PaymentMethod::Update request, we need to include the `BillingAddress` in the payload to be sent to the PSP.

This is required to increase the acceptance rate of the payment method specifically in countries like US.

**Testing**
Ran the integration tests locally for Payment method and billing address was sent along.

### CODE REVIEW

**Review by 0sc — APPROVED** [review]

**Comment by 0sc** [comment]
:shipit: 

Approved with Zappr  Approved with Zappr

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+### Ticket
+<!-- Ticket, issue or anything else that 1. describes the feature and 2. proves its need. -->
+
+### Description
+<!-- You can 1. describe what this pull request implements, 2. give an overview of the proposed solution, 3. add any further comments you might find important, such as why something is missing; whether there are dependencies for this to be merged. Feel free also to add demos (screenshots, videos) if you find that useful. -->
+
+### Checklist
+<!-- Tick with "x" the boxes that apply. You can also fill these out after creating the PR. -->
+* [ ] I've read the [contribution guidelines](CONTRIBUTING.md)
```
Comment (Seklfreak) [inline .github/pull_request_template.md:9]: I don't see `CONTRIBUTING.md` in the PR, is it included somewhere?

File: address.go [hunk 2]
```diff
@@ -25,20 +26,21 @@ type Address struct {
...
-	StreetAddress      string   `xml:"street-address,omitempty"`
-	ExtendedAddress    string   `xml:"extended-address,omitempty"`
-	Locality           string   `xml:"locality,omitempty"`
-	Region             string   `xml:"region,omitempty"`
-	PostalCode         string   `xml:"postal-code,omitempty"`
-	CountryCodeAlpha2  string   `xml:"country-code-alpha2,omitempty"`
-	CountryCodeAlpha3  string   `xml:"country-code-alpha3,omitempty"`
-	CountryCodeNumeric string   `xml:"country-code-numeric,omitempty"`
-	CountryName        string   `xml:"country-name,omitempty"`
+	XMLName            xml.Name
```
Comment (Seklfreak) [inline address.go:31]: why are we removing
```
`xml:"address"`
```
? It seems like we are always injecting it in the code?

**Review by Seklfreak — APPROVED** [review]

File: address.go (hunk 2 above)
Comment (saopayne) [reply address.go:31]: Yeah, that's correct. The full conversation that explains why can be followed here: https://github.com/braintree-go/braintree-go/pull/283#discussion_r272950095

File: .github/pull_request_template.md (hunk 1 above)
Comment (saopayne) [reply .github/pull_request_template.md:9]: Ahh, totally missed that. I will update it. Thanks.

**Comment by ozairasim** [comment]
👍 

Approved with Zappr  Approved with Zappr

**Comment by 0sc** [comment]
:shipit: 

Approved with Zappr  Approved with Zappr

**Review by Seklfreak — APPROVED** [review]

**Comment by Seklfreak** [comment]
👍 

Approved with Zappr  Approved with Zappr

**Comment by 0sc** [comment]
:shipit: 

Approved with Zappr  Approved with Zappr

**Comment by spowani** [comment]
+1

Approved with Zappr  Approved with Zappr

File: init_test.go [hunk 3]
```diff
@@ -78,7 +78,6 @@ func testSubMerchantAccount() string {
 
 func init() {
 	logEnabled := flag.Bool("log", false, "enables logging")
-	flag.Parse()
```
Comment (puellanivis) [inline init_test.go:81]: This isolates the `logEnabled` so it will always be `false`.

File: payment_method_integration_test.go [hunk 4]
```diff
@@ -40,9 +56,52 @@ func TestPaymentMethod(t *testing.T) {
...
+			t.Errorf("Got paymentMethod billing adress country name %#v, want %#v", ba.CountryName, addr.CountryName)
+		}
+	} else {
+		t.Error("paymentMethod should have been a credit card")
+	}
+
 	// Update using different credit card
 	rand.Seed(time.Now().UTC().UnixNano())
 	token := fmt.Sprintf("btgo_test_token_%d", rand.Int()+1)
+	t.Fatal(token)
```
Comment (puellanivis) [inline payment_method_integration_test.go:104]: Why always fail here?

File: init_test.go (hunk 3 above)
Comment (saopayne) [reply init_test.go:81]: This raises an error documented here: https://github.com/golang/go/issues/31859 during initialization.

File: payment_method_integration_test.go (hunk 4 above)
Comment (saopayne) [reply payment_method_integration_test.go:104]: Was added during testing locally for this file. Will remove shortly.

**Pull request merged**